programs/pkey/key_app.c

```diff
@@ -265,8 +265,11 @@ int main( int argc, char *argv[] )
 exit:
 
 #if defined(MBEDTLS_ERROR_C)
-    mbedtls_strerror( ret, buf, sizeof(buf) );
-    mbedtls_printf( "  !  Last error was: %s\n", buf );
+    if ( ret != 0 )
```

Comment:
Just before this check to `ret` there are multiple calls to `mbedtls_mpi_write_file`, none of which check their return code. Can you please expand the error checking to the other library calls?